fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/impl/datagen/FabricDataGenHelper.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.datagen;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint;
+import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.entrypoint.EntrypointContainer;
+
+@ApiStatus.Internal
+public final class FabricDataGenHelper {
+	private static final Logger LOGGER = LogManager.getLogger();
+
+	/**
+	 * When enabled the dedicated server startup will be hijacked to run the data generators and then quit.
+	 */
+	public static final boolean ENABLED = System.getProperty("fabric-api.datagen") != null;
+
+	/**
+	 * Sets the output directory for the generated data.
+	 */
+	private static final String OUTPUT_DIR = System.getProperty("fabric-api.datagen.output-dir");
+
+	/**
+	 * When enabled providers can enable extra validation, such as ensuring all registry entries have data generated for them.
+	 */
+	private static final boolean STRICT_VALIDATION = System.getProperty("fabric-api.datagen.strict-validation") != null;
+
+	/**
+	 * Filter to a specific modid with this property, useful if dependencies also have data generators.
```

Comment:
Use of MOD_ID as separate words in the actual variable name, but modid as one word in the doc is inconsistent. Mod ID should be separate words, with ID capitalized in docs.
```suggestion
	 * Filter to a specific mod ID with this property, useful if dependencies also have data generators.
```